extension/observer/ecsobserver/README.md

```diff
@@ -0,0 +1,330 @@
+# Amazon Elastic Container Service Observer
+
+**Status: beta**
+
+The `ecsobserver` uses the ECS/EC2 API to discover prometheus scrape target from all running tasks and filter them based
+on service names, task definitions and container labels.
+
+## Config
+
+The configuration is based on
+[existing cloudwatch agent ECS discovery](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/ContainerInsights-Prometheus-Setup-autodiscovery-ecs.html)
+. A full collector config looks like the following:
+
+```yaml
+extensions:
+  ecs_observer:
+    refresh_interval: 15s
+    cluster_name: 'Cluster-1'
+    cluster_region: 'us-west-2'
+    result_file: '/etc/ecs_sd_targets.yaml'
+    sevice:
+      - name_pattern: '^retail-.*$'
+    docker_labels:
+      - port_label: 'ECS_PROMETHEUS_EXPORTER_PORT'
+    task_definitions:
+      - job_name: 'task_def_1'
+        metrics_path: '/metrics'
+        metrics_ports:
+          - 9113
+          - 9090
+        arn_pattern: '.*:task-definition/nginx:[0-9]+'
+
+receivers:
+  prometheus:
+    config:
+      scrape_configs:
+        - job_name: "ecs-task"
+          file_sd_configs:
+            - files:
+                - '/etc/ecs_sd_targets.yaml'
+
+processors:
+  batch:
+
+exporters:
+  awsemf:
+
+service:
+  pipelines:
+    metrics:
+      receivers: [ prometheus ]
+      processors: [ batch ]
+      exporters: [ awsemf ]
+  extensions: [ ecs_observer ]
+```
+
+| Name             |           | Description                                                                                      |
+|------------------|-----------|--------------------------------------------------------------------------------------------------|
+| cluster_name     | Mandatory | target ECS cluster name for service discovery                                                    |
+| cluster_region   | Mandatory | target ECS cluster's AWS region name                                                             |
+| refresh_interval | Optional  | how often to look for changes in endpoints (default: 10s)                                        |
+| result_file      | Optional  | path of YAML file for the scrape target results. When enabled, the observer always returns empty |
```

Comment:
Can `result_file` field be mandatory? Otherwise, what's purpose of observer?

Reply:
Changed to mandatory for now,  it's not mandatory in the original PR because receiver creator is supported and the extension just returns the endpoint to observer listener. To reduce the size of PRs I plan to only implement the result file logic.